### DISCUSSION TITLE
System tray icon artwork & animation feedback

### DISCUSSION THREAD

**Discussion by mairin** [podman-desktop/podman-desktop#3154] [Design Feedback]
Hi 👋🏻 !

I've been playing around with updating the system tray icon artwork. ([You can see the WIP PR here](https://github.com/containers/podman-desktop/pull/2952).) One of the main things I wanted to do with updating the system tray icon artwork was to snap the artwork to the pixel grid so it renders more cleanly. See how it looks on GNOME (although, it might be too crisp, I think I might need to thicken the outer border here):

### Before (GNOME)
![Screenshot from 2023-07-07 16-41-24](https://github.com/containers/podman-desktop/assets/799683/8b9daee6-8b9d-4a9c-b138-8c0f7c562e67)

### After (GNOME)
![Screenshot from 2023-07-07 16-41-56](https://github.com/containers/podman-desktop/assets/799683/73a0b8c9-1fb1-4aec-b553-e8d31ade6245)

I have a couple candidates I'm looking at, for macOS specifically, for the animation. Let me know if you have a preference between these or an idea for something better:

### Candidate #1 (macOS)

https://github.com/containers/podman-desktop/assets/799683/e1f7aa7a-472b-4049-8177-0832ec06405e

### Candidate #2 (macOS)

https://github.com/containers/podman-desktop/assets/799683/6fff1543-c4ab-4726-8c30-bfb57ee8c646

**Comment by mairin** [comment]
(Just confirming - yep, the Linux / GNOME tray icon looks a lot nicer with the thicker outline :) )

![Screenshot from 2023-07-07 16-51-16](https://github.com/containers/podman-desktop/assets/799683/ac7403b8-3508-48a8-842f-6867217936f9)

**Reply by benoitf** [reply]
I don't know for the animation. I'm not sure I like both of them :-/ I like some stuff from 1. And some from 2.

Could we animate the outer line as well like something turning with dots instead of lines or some lines disappearing

Maybe I won't like it as well, just ideas.

**Reply by mairin** [reply]
Yeh, it's a good idea to try the outer line, so I'll do that next and see how it turns out!

**Comment by mairin** [comment]
Here's one try, it's 4 frames, I have it at 2x as an animated gif preview below:
![tray-icon-step0Template2](https://github.com/containers/podman-desktop/assets/799683/dd887434-3c3b-4d15-9360-3ef67a02b7fa)

Here it is in situ on the mac:

https://github.com/containers/podman-desktop/assets/799683/6ed09ec9-83fd-402f-92f0-6f5a5c1fb4d4

(there's weird flicky cube things going on in the screen recording that arent there in real life 🤷🏻‍♀️ )

**Reply by blaisep** [reply]
Yes, it's better to indicate the activity by changing the perimeter instead of the inner rectangle. Less distracting because I don't have to figure out what is going on with the rectangle.

**Comment by afbjorklund** [comment]
You need some kind of legacy plug-in to see the system tray icons on GNOME, right? Are you using "AppIndicator", or which one ?

![podman-desktop-fedora-workstation](https://github.com/containers/podman-desktop/assets/10364051/ef43024f-7fea-4784-acd9-c184ebc77591)

(Fedora Workstation 38, Podman Desktop flatpak)

**Reply by mairin** [reply]
Yep, there's 2 or 3 extensions that do it. This is the one I use:

https://extensions.gnome.org/extension/2890/tray-icons-reloaded/

Note, I dont know what size this one sets the icons to. But they gotta be 16, 32, etc px to look good (across all electron apps.) I had a devil of a time trying to figure out how the precisely pixel-grid snapped Podman Desktop icon + the Slack icon looked poorly; I figured out the extension on my computer had been set to display 16px icons at 20px. 🤦‍♀️

**Reply by afbjorklund** [reply]
Trying to install Podman Desktop from the download, or trying to find "tray icons" from within the system both failed.

![podman-desktop-fedora-install](https://github.com/containers/podman-desktop/assets/10364051/32179e96-3928-4ea0-a549-2d355edcc54f)

The easiest was to use the web site instead, and go https://flathub.org/apps/io.podman_desktop.PodmanDesktop

Similar experience for the GNOME extension, I guess (or the minikube extension). Copy/paste URL is the new black?

**Reply by afbjorklund** [reply]
With the extension installed it works, but I guess the icon got a bit squashed (or otherwise showing some big pixels)

![podman-desktop-tray-icon](https://github.com/containers/podman-desktop/assets/10364051/a9cac79b-e1b9-40e0-b1be-9d952262c84e)

**Reply by mairin** [reply]
@afbjorklund Yeh that's the old icon which isn't pixel snapped.... however... it shouldn't look that bad. If you go into the extension settings, switch the icon size to 16px and it'll look way better!

I suspect the flatpak install issues you have are related to a current known bug in flatpak installs in f38 - https://discussion.fedoraproject.org/t/flatpak-install-update-fails-with-need-more-input-or-input-buffer-too-small/84696

(Hopefully it's not a Podman Desktop specific issue?)

**Reply by afbjorklund** [reply]
The tray icon doesn't really serve any purpose on Linux (if not running a Podman Machine or a Kubernetes cluster), anyway.

The main purpose is (probably) to monitor the virtual machine running in the background, or to switch kubernetes context.

**Reply by afbjorklund** [reply]
Changing from the default size (20) to 16 pixels, removes the scaling. Now to try the monochrome icon version...

**Reply by mairin** [reply]
@afbjorklund well its also handy to pull up the window if its minimized 🤷🏻‍♀️

**Reply by afbjorklund** [reply]
But I use the dock for that. Oh well, it might become relevant - if you are on an older system (_with_ tray), and need to use a machine.

Until then you just get a bunch of "green lights" for the Podman socket and for the Docker socket, but nothing is really "running"...

Or at least it wasn't, until Podman Desktop started another daemon (like `podman.service`)

* https://github.com/containers/podman-desktop/issues/2903

**Reply by afbjorklund** [reply]
Actually it looked even worse on the classic system,  since it is "light" and not "dark" (and got some funky wraparound):

![podman-desktop-centos-mate](https://github.com/containers/podman-desktop/assets/10364051/2b1bac6d-1dd9-4186-b5a9-94cec99f971f)

This podman did not have any remote socket (it still used varlink). Here is a zoom-in on the system tray: (it's [MATE](https://mate-desktop.org/))

![mate-dark](https://github.com/containers/podman-desktop/assets/10364051/6b024eb6-bec6-42f6-b3ad-33a9b39b9ef7)

**Reply by mairin** [reply]
Ooh yikes. I will look into this, thanks for checking!!

**Reply by afbjorklund** [reply]
It will probably fall under "unsupported", anyway

**Comment by NeilHanlon** [comment]
Pick number two :)

**Comment by KyleRConway** [comment]
On animation: This is a quick, imperfect mock-up ― could you have a pulse go from left to right and then circle back around? I'm not sure what the limitations are here (color, frames, etc.), but here's a mock-up either way.

https://github.com/containers/podman-desktop/assets/5899012/88e529d8-5802-4f6c-bbfa-78af0d18c206

**Reply by KyleRConway** [reply]
Here's another that sort of moves through the hexagon as a flash.

Multi-page Inkscape file: ![tray_icon_5_flash](https://github.com/containers/podman-desktop/assets/5899012/9325a34f-ef46-4931-ab3a-30bbf4cbf1c8)

Animation Gif: ![flash](https://github.com/containers/podman-desktop/assets/5899012/49a304db-ce6b-4d04-9001-3ac9b7814197)

**Reply by mairin** [reply]
Eeeeee this is awesome, Kyle! The svg makes it easy to try to implement too. When I get back to my laptop I'll give it a go.

**Reply by mairin** [reply]
https://github.com/containers/podman-desktop/assets/799683/72d92699-4068-4144-8bc5-bb28a2bcf401

This is how it came out in the app... I figured out how to add the extra frame. I think animating the lights on the box + the outer frame is a good idea. I think maybe I'll add some alpha gradients instead of the clean cuts to see how it looks. 

(also the weird floaty box is a compression artifact from the macos screen recorder... 🤷🏻‍♀️ )

**Reply by mairin** [reply]
I wish I knew how to get clean screen recordings off of a mac :( they always seem to get these compression artifacts! Anyway :) Here it is with the gradient effect


https://github.com/containers/podman-desktop/assets/799683/6ad16454-48b0-4634-ad57-32849a0bf9b2

**Reply by KyleRConway** [reply]
I do like the alpha!

**Comment by blaisep** [comment]
Thanks to all of you for your service!!

**Comment by afbjorklund** [comment]
What does it actually mean, when the icon is "spinning" ? Is it like plymouth, trying to show that it is starting or stopping ?

----

Trying to replace the log with a progress bar for Lima, but it is not included in any release just yet: https://github.com/lima-vm/lima/issues/1311

```
Starting "podman"
Waiting ....
READY.
```

Once it can output progress events as JSON, it will be possible to have a more detailed start up (currently just "spinning")

**Comment by deboer-tim** [comment]
> What does it actually mean, when the icon is "spinning" ? Is it like plymouth, trying to show that it is starting or stopping ?

Correct, it animates when a provider or connection is starting or stopping. I took a look at the code to confirm - it currently has a initialized/empty, error, updating, and ready/default states, and the updating state cycles through 4 steps changing every 1000ms.

Mairin, I see no reason why we couldn't change this, e.g. have separate starting/stopping animations (or run the same animation in reverse), or change the # of steps/speed (e.g. 6 steps changing every 400ms, or 8 steps and 500ms, or ...).

To me the animations could use a couple more steps to be smooth, but we could always try something to make sure we like it and then add intermediate steps (i.e. double the # of steps and modify the speed). Personally I prefer it when I can tell the difference between starting and stopping, which usually means starting is some flavour of up/right/clockwise/fill/brighten, and then we can reverse it for stopping.

**Reply by afbjorklund** [reply]
It would need better support from `podman machine`, and probably in the end also from Fedora CoreOS, to show more details.

Currently it just does major operations*, without feedback... And even when there is progress, it is "hidden" in the logs tab.

\* Downloading, Decompressing

Anyway, not for display in systray.

**Reply by afbjorklund** [reply]
Added an issue for the decompression.

* https://github.com/containers/podman/issues/19240

But it is still "cut off", in the PD log tab:

```
Downloading VM image: fedora-coreos-38.20230709.2.0-qemu.x86_64.qcow2.xz: ...  
Downloading VM image: fedora-coreos-38.20230709.2.0-qemu.x86_64.qcow2.xz: ...  
Downloading VM image: fedora-coreos-38.20230709.2.0-qemu.x86_64.qcow2.xz: ...  
Downloading VM image: fedora-coreos-38.20230709.2.0-qemu.x86_64.qcow2.xz: ...  
Downloading VM image: fedora-coreos-38.20230709.2.0-qemu.x86_64.qcow2.xz: ...  
Downloading VM image: fedora-coreos-38.20230709.2.0-qemu.x86_64.qcow2.xz: ...  
Downloading VM image: fedora-coreos-38.20230709.2.0-qemu.x86_64.qcow2.xz: ...  
Downloading VM image: fedora-coreos-38.20230709.2.0-qemu.x86_64.qcow2.xz: ...  
```

**Comment by mairin** [comment]
Quick update, Paul Power (an intern on the community design team) came up with this idea - it has a lot of frames so I have to poke at it to get a more condensed yet still smooth version:

![podman-tray-icon-2](https://github.com/containers/podman-desktop/assets/799683/e3f16d33-38ec-4325-a99d-18ef57361ee8)

**Comment by mairin** [comment]
btw if anybody wants to try the WIP I have so far (working on macOS and Linux, Windows is a little iffy), this is the PR:

https://github.com/containers/podman-desktop/pull/2952

**Comment by rujutashinde** [comment]
closing this discussion as it is either implemented or not needed
